python/poetry/Portfile

```diff
@@ -42,6 +42,9 @@ foreach pv {39 38 37} {
     }
 }
 
+depends_build-append \
+    port:py${python.version}-setuptools
```

Comment:
Thanks @reneeotten for updating this 👍.

My understanding is that `tool.poetry.scripts` is equivalent to `console_scripts`, so wouldn't this be a library dependency?

From [pyproject.toml](https://github.com/python-poetry/poetry/blob/06dfddece6d8865f12324664e74959884ba3ea27/pyproject.toml#L66-L67):

```
[tool.poetry.scripts]
poetry = "poetry.console.application:main"
```

Reply:
quite a few Python packages these days do something like this (see, for example, `/opt/local/bin/poetry'):
```
try:
    from importlib.metadata import distribution
except ImportError:
    try:
        from importlib_metadata import distribution
    except ImportError:
        from pkg_resources import load_entry_point
```

so it will try and use `importlib-metadata` if available (and that's an dependency listed in `setup.py` by upstream) to run. It will only go back to using `pkg_resources` (i.e., `setuptools`) if that is not available. 

I added `py-importlib-metadata` as a dependency so `setuptools` isn't needed at runtime in this case (this will probably happen in more ports going forward so the dependency type of `setuptools` might need to be reviewed in such cases where also `py-importlib-metadata` is a dependency).

Reply:
Ah I see, thanks for explaining @reneeotten.